org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/StandardProjectsManager.java

```diff
@@ -215,6 +220,16 @@ public void fileChanged(String uriString, CHANGE_TYPE changeType) {
 		if (resource == null) {
 			return;
 		}
+		URI settingsUri = preferenceManager.getPreferences().getSettingsAsURI();
+		if (settingsUri != null) {
+			URI uri = JDTUtils.toURI(uriString);
+			if (uri != null && uri.equals(settingsUri) && JavaLanguageServerPlugin.getInstance().getProtocol() != null) {
```

Comment:
I  noticed here that the `uri` coming from the client had the form `file:///....` while the `settingsUri` was `file:/...`. I wasn't able to delete the global settings file because of another issue but I would check if this equals gets satisfied when needed.

Reply:
You can try the following test:
```
@Test
	public void testUri() throws URISyntaxException {
		URI uri1 = new URI("file:/test");
		URI uri2 = new URI("file:///test");
		assertTrue(uri1.equals(uri2));
	}
```

Reply:
You're right. It works :)